### PR TITLE
Fix endless loop in demo replay, allow skip before game start

### PR DESCRIPTION
**Pull request by eun-ice**
## Summary

1. Allow skipping a demo replay before game start by removing the pregame guard from SkipTo.
2. When skipping during the countdown, start the game immediately after the skip completes so the countdown ends.
3. Fix the endless skip loop by advancing modGameTime with GetNextDemoReadTime instead of GetModGameTime.

## Testing

Manually verified with a forced 10-second countdown and pressing Skip during the countdown (around 8s)

-   without this PR: Skip does not stop countdown, game is paused after countdown
-   with the StartGame(true) change but without GetModGameTime() fix: countdown does not stop, engine freezes
 -  with the GetModGameTime() fix but without StartGame(true): Countdown is no longer shown, but game shows "game paused" until the countdown time is elapsed and then replay resumes normally
 -  with both StartGame(true) and GetModGameTime() fix: Everything works as expected, countdown stops, demo playback starts immediately

Addresses the replay-skip portion of #3209.

## AI disclosure

OpenAI Codex assisted with diagnosing the endless loop, implementing the change, and drafting this pull request. The behavior was manually tested and confirmed by the contributor.

### CODE REVIEW

**Comment by eun-ice** [comment]
@sprunk Once again, your intuition was correct: the packet handling was only fixing a symptom.

The actual problem was an endless loop in SkipTo:

  ```cpp
        while (SendDemoData(targetFrameNum)) {
                gameTime = GetDemoTime();
                modGameTime = demoReader->GetModGameTime() + 0.001f;

  ```

If GetModGameTime returns 0 and the first demo packet’s scheduled read time is larger than 0.001f this loops endlessly.

I hope this is now the proper fix.

File: rts/Net/GameServer.cpp [hunk 1]
```diff
@@ -431,6 +431,8 @@ void CGameServer::SkipTo(int targetFrameNum)
 
 	if (udpListener != nullptr)
 		udpListener->Update();
+	if (wasPregame)
+		StartGame(true); // this skips the countdown
```
Comment (sprunk) [inline rts/Net/GameServer.cpp]: is this needed? if we were pregame then one of the packets sent should be the "start playing" packet

i'm also not sure what happens if the start packet is sent in the middle of a game (where we skipped to), perhaps if it's needed try seeing if it can be sent before the loop and not after?

Comment (eun-ice) [reply rts/Net/GameServer.cpp]: It is required see added test protocol in the description.

I moved the StartGame(true) below the loop because I thought it would fix an issue: If click "skip" with a 10 second countdown showing "8", then the countdown would briefly display a lower number (e.g. "3") before starting the game.

My guess was this is due to the loop processing the packets and moving it later seemed to fix it. 
Upon retesting the same artifact remains regardless of whether StartGame(true) is top or bottom, so no need to move it down. This is not critical at all, it's just visible for one frame or so.

I moved it up.

**Comment by eun-ice** [comment]
@sprunk after even more manual testing, I think this is good now. I still would prefer if someone would test remote replays with this.

**Review by sprunk — APPROVED** [review]
Looks alright, I have no idea what sort of edge cases may break but maybe we will find out sooner or later either way.

**Comment by eun-ice** [comment]
Thank you very much. I got a local remote replay to work and that still worked. However, the skip button and the game speed buttons did not have an effect, but I think that has nothing to do with my changes, the server sees to have rejected them always with `if (serverCommand && players[a].isLocal)`
